web/src/compose_recipient.ts

```diff
@@ -256,17 +249,13 @@ function item_click_callback(event: JQuery.ClickEvent, dropdown: tippy.Instance)
 function get_options_for_recipient_widget(): Option[] {
     const options: (Option | DirectMessagesOption)[] =
         stream_data.get_options_for_dropdown_widget();
-
+    const recipients = compose_pm_pill.get_user_ids_string();
     const direct_messages_option = {
         is_direct_message: true,
         unique_id: compose_state.DIRECT_MESSAGE_ID,
         name: $t({defaultMessage: "Direct message"}),
     };
-
-    if (
-        realm.realm_private_message_policy ===
-        settings_config.private_message_policy_values.by_anyone.code
-    ) {
+    if (user_can_send_direct_message(recipients)) {
         options.unshift(direct_messages_option);
```

Comment:
I don't quite understand this logic; can you explain it? `compose_pm_pill.get_user_ids_string` will very possibly be the empty string if you've currently got a stream in the compose box.

Reply:
I assume you mean "are not disabled for all users" :)

Reply:
No, it should be at the end of the list when direct messages are disabled, right?

Reply:
Ahh, OK, yes that makes sense.